### ISSUE TITLE
False positive `Lint/LiteralsComparison` when comparing a generic type in a macro

### ISSUE THREAD

**Issue by DanielGilchrist** [crystal-ameba/ameba#318]
Hello and thanks a bunch for this shard!

I recently came across a false positive with the `Lint/LiteralsComparison` rule. When using the `==` operator comparing `TypeNode`s that could be equivalent, ameba complains

```crystal
class Klass(T)
  def initialize
    {% if T == Nil %}
      puts "T is Nil"
    {% end %}
  end
end

Klass(Nil).new
# "T is Nil" is outputted during compile time
```

Output running ameba
<img width="564" alt="image" src="https://user-images.githubusercontent.com/13454550/206465398-0e74b1d2-2fbf-46c1-8b0c-7f31167b95fe.png">

**Comment by Sija** [comment]
Yup, fix is on the way :)

**Comment by Sija** [comment]
Btw, this should be fixed in version `1.3.1`.

**Comment by DanielGilchrist** [comment]
> Btw, this should be fixed already in version `1.3.1`.

It looks like I was using an older version of ameba while running locally as I hadn't recompiled the ameba binary 🤦‍♂️ Have confirmed the issue is fixed on 1.3.1 and can close this issue. Thank you!
